### ISSUE TITLE
App Crashed while Converting Images to PDF or Text to PDF in API 29

### ISSUE THREAD

**Issue by Arsalankhan1212** [Swati4star/Images-to-PDF#973]
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
        at androidx.loader.content.ModernAsyncTask$3.done(ModernAsyncTask.java:164)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:929)
     Caused by: android.database.sqlite.SQLiteException: near "GROUP": syntax error (Sqlite code 1 SQLITE_ERROR): , while compiling: SELECT _id, bucket_id, bucket_display_name, _data, COUNT(*) AS count FROM files WHERE ((is_pending=0) AND (is_trashed=0) AND (volume_name IN ( 'external_primary' ))) AND (((media_type=? OR media_type=?) AND _size>0) GROUP BY (bucket_id)) ORDER BY datetaken DESC, (OS error - 2:No such file or directory)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:184)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
        at android.content.ContentProviderProxy.query(ContentProviderNative.java:423)
        at android.content.ContentResolver.query(ContentResolver.java:955)
        at android.content.ContentResolver.query(ContentResolver.java:891)
        at androidx.core.content.ContentResolverCompat.query(ContentResolverCompat.java:81)
        at androidx.loader.content.CursorLoader.loadInBackground(CursorLoader.java:63)
        at com.zhihu.matisse.internal.loader.AlbumLoader.loadInBackground(AlbumLoader.java:97)
        at com.zhihu.matisse.internal.loader.AlbumLoader.loadInBackground(AlbumLoader.java:33)
        at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
        at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
```

**Comment by IsraaHamed1195566** [comment]
I want to try solve it

**Comment by IsraaHamed1195566** [comment]
did you check your android API level ?

**Comment by kashifkhanbaloch** [comment]
update the image picker library in build.gradle file

**Comment by Arsalankhan1212** [comment]
I've already solved all the crash
Most crashes was about the file path in API 29 and i solved it by using
this file picker library 'com.github.hedzr:android-file-chooser:1.2.0'
but its limitation is, it only work upto android 10 (api level 29) and not
on android 11

And also some crashes occur, i don't exactly remember the crash but mostly
the app crashed as it was unable to create a directory on external storage
in api level 29 and above

**Comment by czlucius** [comment]
Can I still work on this issue?

Thanks.

**Comment by Arsalankhan1212** [comment]
Yes, please do, and also upgrade code to latest SDK i.e. Api Level 30

**Comment by Swati4star** [comment]
yes @czlucius 👍

**Comment by errohitdev** [comment]
Anyone resolve this issue??

**Comment by czlucius** [comment]
I couldn't reproduce this issue on Android 10

**Comment by avneeshkumar1** [comment]
Yes i have fixed this issue and its working in android 11 also.

**Comment by errohitdev** [comment]
can you share your source code at allmistree@gmail.com

**Comment by Rautarchit001** [comment]
Hii everyone 

we just need to update the image picker library, can anyone say me how to update that in this git project?

**Comment by Rautarchit001** [comment]
Any update on this?

**Comment by Rautarchit001** [comment]
This issue resolve my updating image picker library but now there is another isuue after picking/selection image does not show in preview and also does not create pdf.

**Comment by codegsaini** [comment]
@Rautarchit001 I updated to API 30 and not found any issue yet. Can you please tell me more about where you encountered error after updating to API 29.

**Comment by Rautarchit001** [comment]
After updating clean project and app from installed device and emulator then build and check it will not create pdf

**Comment by codegsaini** [comment]
> App doesn't crash but don't create pdf

Are you talking about **Images to PDF** feature.

**Comment by Rautarchit001** [comment]
Yes

**Comment by codegsaini** [comment]
Please mention your device's android version in which you are testing app.

**Comment by Rautarchit001** [comment]
One Plus 6T - Android:- 10
Google Pixel X Emulator Android:- 9&10

**Comment by codegsaini** [comment]
Please confirm : If you encountered this issue in android 9 (or below android 9 )!

**Comment by Rautarchit001** [comment]
Yes in below version of Android 9 preview was working but pdf is not creating.

**Comment by Arsalankhan1212** [comment]
> Yes i have fixed this issue and its working in android 11 also.
> I have converted full code to kotlin and reduce app size and fix multiple issues.

Can you share your kotlin version code that target android 11?

**Comment by codegsaini** [comment]
@Arsalankhan1212 this issue is already fixed in pull request #994, Please consider reviewing it.
